lib/spack/spack/version.py

```diff
@@ -153,6 +157,27 @@ def lowest(self):
     def highest(self):
         return self
 
+    def isnumeric(self):
+        """Tells if this version is numeric (vs. a non-numeric version).  A
+        version will be numeric as long as the first section of it is,
+        even if it contains non-numerica portions.
+
+        Some numeric versions:
+            1
+            1.1
+            1.1a
+            1.a.1b
+        Some non-numeric versions:
+            develop
+            system
+            myfavoritebranch
+        """
+        return isinstance(self.version[0], int)
```

Comment:
Looks like you're right:

http://stackoverflow.com/questions/8203336/difference-between-int-and-numbers-integral-in-python